Lib/test/test_unittest/testmock/testpatch.py

```diff
@@ -1979,5 +1979,86 @@ def test(): pass
             test()
 
 
+    def test_method_patched_by_subclass1(self):
+        @patch('%s.something' % __name__, 1)
+        class Foo:
+            def test(self):
+                return something
+
+        @patch('%s.something' % __name__, 2)
+        class Bar(Foo):
+            pass
+
+        self.assertEqual(Foo().test(), 1)
+        self.assertEqual(Bar().test(), 2)
+
+
+    def test_method_patched_by_subclass2(self):
+        class Foo:
+            def test(self):
+                return something
+
+        @patch('%s.something' % __name__, 2)
+        class Bar(Foo):
+            pass
+
+        self.assertEqual(Bar().test(), 2)
+
+
+    def test_method_patched_by_subclass3(self):
+        class Foo: ...
+
+        class Mixin:
+            def test(self):
+                return something
+
+        @patch('%s.something' % __name__, 2)
+        class Bar(Foo, Mixin):
+            pass
+
+        self.assertEqual(Bar().test(), 2)
+
+
+class AsyncPatchTest(unittest.IsolatedAsyncioTestCase):
+    async def test_async_method_patched_by_subclass1(self):
+        @patch('%s.something' % __name__, 1)
+        class Foo:
+            async def test_async(self):
+                return something
+
+        @patch('%s.something' % __name__, 2)
+        class Bar(Foo):
+            pass
+
+        self.assertEqual(await Foo().test_async(), 1)
+        self.assertEqual(await Bar().test_async(), 2)
+
+
+    async def test_async_method_patched_by_subclass2(self):
+        class Foo:
+            async def test_async(self):
+                return something
+
+        @patch('%s.something' % __name__, 2)
+        class Bar(Foo):
+            pass
+
+        self.assertEqual(await Bar().test_async(), 2)
+
+
+    async def test_async_method_patched_by_subclass3(self):
+        class Foo: ...
+
+        class Mixin:
+            async def test_async(self):
+                return something
+
+        @patch('%s.something' % __name__, 2)
+        class Bar(Foo, Mixin):
+            pass
+
+        self.assertEqual(await Bar().test_async(), 2)
```

Comment:
Please, also test deep inheritance with at least three levels.

Reply:
I've made the test classes in `test_method_patched_by_subclass1` one level deeper then. I think that would be enough.